### PR TITLE
Add Heatmap support

### PR DESCRIPTION
**Pull request by jrichardlai**
Support Heatmap:

out
- Uses GoogleMap api for Android.
- Uses [DTMHeatmap](https://github.com/dataminr/DTMHeatmap) for iOS.

Api: 

```
points = Array<WeightedLatLng>

type WeightedLatLng {
  latitude: Number,
  longitude: Number,
  weight: Number, // optional
}


<MapView.Heatmap points={points} />

```

Note: Sorry about the WIP commits, need to be squashed before merging into master.

### CODE REVIEW

**Comment by jrichardlai** [comment]
@lelandrichardson @christopherdro thoughts on that?

**Comment by christopherdro** [comment]
@jrichardlai Wow, this looks great!

I'm just not quite sure if using a third party library would be best, but thats for @lelandrichardson to decide. Maybe this can be added as an extension?

**Comment by jrichardlai** [comment]
@christopherdro yeah agree was not sure what would be best for this. What are you thinking of when you say "added as an extension"?

**Comment by christopherdro** [comment]
Like another npm module that adds this feature to react-native-maps.

**Comment by lelandrichardson** [comment]
This is super freakin cool. I need to think about what this means for us though. Will take a look as soon as i can, but just FYI I'm going on vacation for ~2 weeks on friday. Will try to look at this beforehand.

**Comment by felipecsl** [comment]
@lelandrichardson vacations? ohnoessss hahah, enjoy man!

File: android/lib/src/main/java/com/airbnb/android/react/maps/AirMapHeatmapManager.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import com.google.android.gms.maps.model.Marker;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.annotation.Nullable;
+
+public class AirMapHeatmapManager extends ViewGroupManager<AirMapHeatmap> {
+    public AirMapHeatmapManager(ReactApplicationContext reactContext) {
+        super();
```
Comment (felipecsl) [inline android/lib/src/main/java/com/airbnb/android/react/maps/AirMapHeatmapManager.java]: you can omit this

File: android/lib/src/main/java/com/airbnb/android/react/maps/AirMapHeatmap.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+import java.util.ArrayList;
+
+public class AirMapHeatmap extends AirMapFeature {
+
+    private TileOverlayOptions heatmapOptions;
+    private TileOverlay heatmap;
+    private HeatmapTileProvider heatmapTileProvider;
+
+    private ArrayList<WeightedLatLng> points;
```
Comment (felipecsl) [inline android/lib/src/main/java/com/airbnb/android/react/maps/AirMapHeatmap.java]: use `List<WeightedLatLng>` instead

**Comment by matlin** [comment]
Is this still being considered as an addition? I guess it's only a matter of time before heatmap support is added to iOS and the DTMHeatmap isn't necessary anymore.

**Comment by kanekotic** [comment]
this seems a cool functionality :)

**Comment by proProbe** [comment]
is this still in the pipeline or has it been rejected?

**Comment by DVassilev** [comment]
@lelandrichardson Had a chance to review it?

**Comment by tuantle** [comment]
Really need this. Hoping to get this functionality soon!!!
+1

**Comment by zolrath** [comment]
Would loooove to see this feature!

**Comment by rococtz** [comment]
Hi, is this PR still active?

**Comment by p3teryan** [comment]
jrichardlai, My guys are anxious to see your update so it can be merged. Please pretty Please!

**Comment by christopherdro** [comment]
Merge conflicts will need to resolved and re tested before getting this feature in.

Perhaps someone would like to help out with that?

**Comment by mickadoua** [comment]
today i tried to install and used, but ... Red Screen of dead


```
npm uninstall react-native-maps --save
npm install https://github.com/airbnb/react-native-maps.git#pull/240/head --save
```
my app cant be launched

**Comment by hubConnect** [comment]
For those wanting to add this feature into their project, an updated version of this pull request is available #1549

**Comment by christopherdro** [comment]
Closing for #1549

**Pull request closed without merging**